ding/framework/middleware/functional/data_processor.py

```diff
@@ -45,6 +46,43 @@ def _push(ctx: "OnlineRLContext"):
     return _push
 
 
+def buffer_saver(cfg: EasyDict, buffer_: Buffer, every_envstep=1000, replace=False):
+    """
+    Overview:
+        Save current buffer data.
+    Arguments:
+        - cfg (:obj:`EasyDict`): Config.
+        - buffer (:obj:`Buffer`): Buffer to push the data in.
+        - every_envstep (:obj:`int`): save at every env step.
+        - replace (:obj:`str`): Whether replace the last file.
+    """
+
+    buffer_saver_counter = 0
+
+    def _save(ctx: "OnlineRLContext"):
+        """
+        Overview:
+            In ctx, `ctx.env_step` should not be None.
+        Input of ctx:
+            - env_step (:obj:`int`): env step.
+        """
+        nonlocal buffer_saver_counter
+        if ctx.env_step is not None:  # each data in buffer is a episode
+            if ctx.env_step >= every_envstep * buffer_saver_counter:
+
+                buffer_saver_counter += 1
```

Comment:
counter should be updated outside the if-else branch

Reply:
fixed.